### ISSUE TITLE
API errors not failing GH validation

### ISSUE THREAD

**Issue by iloveeclipse** [eclipse-platform/eclipse.platform.ui#2486]
> But why they are both "green" ? Shouldn't they have red crosses?

Because platform.ui has chosen to make them green:
https://github.com/eclipse-platform/eclipse.platform.ui/blob/c1c681ada78722481d2b01a0790bf87f966975e4/Jenkinsfile#L33

>  Is there a new version available? If yes, could you please update aggregator to use it?

There was some issues with deadlocks but no one cared enough to analyses the root cause so an older API-Tools version was used in the meanwhile. I have now created a PR to configure the changed behavior now but it is not yet released:
- https://github.com/eclipse-tycho/tycho/pull/4295

_Originally posted by @laeubi in https://github.com/eclipse-platform/eclipse.platform.ui/issues/2476#issuecomment-2453878664_

**Comment by iloveeclipse** [comment]
@laeubi : I'm not sure which part of the mentioned line is responsible for showing "green" vs "red" Github checks shown on GH UI.
Is this `publishAllIssues:false` or `ignoreQualityGate:true` or both or something else?

**Comment by laeubi** [comment]
> Is this `publishAllIssues:false` or `ignoreQualityGate:true` or both or something else?

- `publishAllIssues` = should all (including existing) be shown in Github as code annotations
- `ignoreQualityGate` = If true, then the result of the quality gate is ignored when selecting a reference build. This option is disabled by default, so a failing quality gate will be passed from build to build until the original reason for the failure has been resolved.

**Comment by iloveeclipse** [comment]
@laeubi : thanks, but sorry, I still don't understand which setting causes the failed API check to be shown as green instead as red? Could you please simply fix whatever you think is broken and push a PR that would "enable" github validation to fail on failed API check?

**Comment by laeubi** [comment]
I don't know what is the desired outcome, when I started with this I let the check fail / become "red" whenever something new was detected, then several iterations / changes where performed e.g. now we use "DELTA" what mean if you solve one problem you are allowed to add one new without the checks to become red.

As long as warnings / reported problems are only seen as something that gets into peoples way to merge there PR as soon as possible instead of something we need to fix/improve I honestly don't see any "right" settings.

This is for example how PDE uses the checks what seem to work (for the PDE project) since a while:
https://github.com/eclipse-pde/eclipse.pde/blob/f3ba0239012344e49638f28431ff4340c1eeeb47/Jenkinsfile#L42

**Comment by iloveeclipse** [comment]
> I don't know what is the desired outcome

API **errors** are not **warnings**, they should **always** fail the build!

**Comment by laeubi** [comment]
> API errors are not warnings, they should always fail the build!

As said this is likely due to a bug that the build do not _fails immediately_. But still these are (additionally) reported as build warnings / errors from the maven parser.

**Comment by jukzi** [comment]
> API **errors** are not **warnings**, they should **always** fail the build!

API errors on master should not fail the jenkins build (otherwise it would not be found as baseline) but mark it unstable. They should fail the GitHub validation of PRs tough.

**Comment by jukzi** [comment]
i.e. it can be build, but should not be submitted

**Comment by sratz** [comment]
> i.e. it can be build, but should not be submitted

Agreed, there should only be one red X to look at, not two (the actual build and the dedicated validation) with the same issues.

**Comment by iloveeclipse** [comment]
Guys, there should be **at least one** red X, currently everything is green, and that is the main problem.

Whoever understands what is the root cause (tycho bug or wrong jenkins file), please provide fix for that.

Note, **if** this is Jenkins file to blame, it seem to be very similar on all other platform repos **except** PDE, so it should be fixed overall.

**Comment by jukzi** [comment]
@iloveeclipse maybe you can start by demonstrating the error? My two attempts failed.

**Comment by iloveeclipse** [comment]
> @iloveeclipse maybe you can start by demonstrating the error? My two attempts failed.

See https://github.com/eclipse-platform/eclipse.platform.ui/pull/2476#issuecomment-2453853847 and all conversation above.

**Comment by jukzi** [comment]
> See [#2476 (comment)](https://github.com/eclipse-platform/eclipse.platform.ui/pull/2476#issuecomment-2453853847) and all conversation above.

I explained that one in https://github.com/eclipse-platform/eclipse.platform.ui/pull/2476#issuecomment-2453968380 ... not an Issue just a bad coincidence which is hard to get rid of with the tools we have.
